autoload/codi.vim

```diff
@@ -186,6 +186,34 @@ function! s:user_au(au)
   endif
 endfunction
 
+" If percent / float width, calculate based on buf width
+" If result leaves less than |minwidth| / 20 columns of working space,
+" return a width that leaves |minwidth| / 20 columns to write code in
+" Else, return absolute width as given
+function! s:pane_width()
+  let width = s:get_opt('width')
+  let width = type(width) == type('') ? width : string(width)
+
+
+  if match(width, '[.%]') > -1
+    let raw          = str2float(substitute(width, '%', '', 'g'))
+    let clamped      = raw > 100.0 ? 100.0 : (raw < 0.0 ? 0.0 : raw)
+    let min_width    = exists('&winwidth') ? &winwidth : 20
```

Comment:
I don't think there will be a case where `winwidth` is not defined, as it is a built-in option.